starboard/shared/starboard/media/iamf_util_test.cc

```diff
@@ -0,0 +1,235 @@
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+  codec_param = "IAMF.000.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Invalid additional profile value
+  codec_param = "iamf.000.999.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Invalid primary profile value
+  codec_param = "iamf.999.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Invalid leading codec param
+  codec_param = "iacb.000.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Invalid length for Opus substream
+  codec_param = "iamf.000.000.Opus.40.2";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Invalid length for AAC-LC substream
+  codec_param = "iamf.000.000.mp4a";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Invalid param for AAC-LC substream
+  codec_param = "iamf.000.000.mp4a.40.3";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Invalid param for AAC-LC substream
+  codec_param = "iamf.000.000.mp4a.40.20";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Too many delimiting periods
+  codec_param = "iamf.000.000..Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Leading delimiting period
+  codec_param = ".iamf.000.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Trailing delimiting period
+  codec_param = "iamf.000.000.Opus.";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // No delimiting period between codec param string and primary profile
+  codec_param = "iamf000.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Invalid primary profile length
+  codec_param = "iamf.00.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+  codec_param = "iamf.0000.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Invalid additional profile length
+  codec_param = "iamf.000.00.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+  codec_param = "iamf.000.0000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Letters in primary profile value
+  codec_param = "iamf.0aa.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+  codec_param = "iamf.xxx.000.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Letters in additional profile value
+  codec_param = "iamf.000.0aa.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+  codec_param = "iamf.000.xxx.Opus";
+  util = IamfMimeUtil(codec_param);
+  EXPECT_FALSE(util.is_valid());
+
+  // Letter in primary profile value (uppercase letter "O" instead of the digit
+  // 0)
```

Comment:
nit: You may replace 'O' with 'a' to shorten the comment.  But please feel free to keep as is.

Reply:
Done